cmd/antrea-agent/agent.go

```diff
@@ -137,7 +137,11 @@ func run(o *Options) error {
 
 	enableAntreaIPAM := features.DefaultFeatureGate.Enabled(features.AntreaIPAM)
 	enableBridgingMode := enableAntreaIPAM && o.config.EnableBridgingMode
-	enableNodePortLocal := features.DefaultFeatureGate.Enabled(features.NodePortLocal) && o.config.NodePortLocal.Enable
+	enableNodePortLocal := o.config.NodePortLocal.Enable
+	if !features.DefaultFeatureGate.Enabled(features.NodePortLocal) {
+		enableNodePortLocal = false
+		klog.InfoS("Feature gate `enableNodePortLocal` is deprecated, please use option `nodePortLocal.enable` to disable NodePortLocal")
```

Comment:
The name of the FeatureGate is `NodePortLocal`, not `enableNodePortLocal`

Reply:
Move the feature check block to `cmd/antrea-agent/options.go` now. A new function `validateNodePortLocalConfg` is added to check the NodePortLocal configuration.